### PR TITLE
Make rest-api optional feature ...

### PR DESCRIPTION
**Pull request by milenkovicm**
... disabled by default

relates to  #1067

# Which issue does this PR close?

Closes #.

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: ballista/scheduler/src/scheduler_server/mod.rs [hunk 1]
```diff
@@ -155,7 +156,7 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> SchedulerServer<T
     pub fn running_job_number(&self) -> usize {
         self.state.task_manager.running_job_number()
     }
-
+    #[allow(dead_code)] // used by rest-api feature
```
Comment (andygrove) [inline ballista/scheduler/src/scheduler_server/mod.rs:159]: can we not do `#[cfg(feature = "rest-api")]` here?

Comment (milenkovicm) [reply ballista/scheduler/src/scheduler_server/mod.rs:159]: we absolutluy could, will fix

File: ballista/scheduler/src/scheduler_server/mod.rs [hunk 2]
```diff
@@ -64,6 +64,7 @@ pub struct SchedulerServer<T: 'static + AsLogicalPlan, U: 'static + AsExecutionP
     pub start_time: u128,
     pub state: Arc<SchedulerState<T, U>>,
     pub(crate) query_stage_event_loop: EventLoop<QueryStageSchedulerEvent>,
+    #[allow(dead_code)] // used by rest-api feature
```
Comment (andygrove) [inline ballista/scheduler/src/scheduler_server/mod.rs:67]: Are the remaining uses of `#[allow(dead_code)]` still necessary?

Comment (milenkovicm) [reply ballista/scheduler/src/scheduler_server/mod.rs]: they are not, my bad, should be fixed now

**Comment by mmooyyii** [comment]
When rest-api is disabled and I request "http://127.0.0.1:50050/api/jobs" or others api. it will return 200 ok with empty body. 
maybe it should return 400 and some error message?

**Comment by milenkovicm** [comment]
> When rest-api is disabled and I request "http://127.0.0.1:50050/api/jobs" or others api. it will return 200 ok with empty body. maybe it should return 400 and some error message?

issue created https://github.com/apache/datafusion-ballista/issues/1223

**Pull request merged**